Userland/Libraries/LibHID/ReportParser.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2025, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/ByteReader.h>
+#include <AK/Endian.h>
+#include <AK/Function.h>
+#include <AK/MemoryStream.h>
+#include <LibHID/ReportDescriptorParser.h>
+#include <LibHID/ReportParser.h>
+
+namespace HID {
+
+ErrorOr<void> parse_input_report(ParsedReportDescriptor const& report_descriptor, ReadonlyBytes report_data, Function<ErrorOr<IterationDecision>(Field const&, i64)> callback)
+{
+    u8 report_id = 0;
+    if (report_descriptor.uses_report_ids) {
+        if (report_data.size() < 1)
+            return Error::from_string_view_or_print_error_and_return_errno("Report is too small"sv, EINVAL);
+
+        report_id = report_data[0];
+    }
+
+    auto maybe_report = report_descriptor.input_reports.get(report_id);
+    if (!maybe_report.has_value())
+        return Error::from_string_view_or_print_error_and_return_errno("Invalid Report ID"sv, EINVAL);
+
+    for (auto const& field : maybe_report->fields) {
+        // 8.4 Report Constraints: An item field cannot span more than 4 bytes in a report. For example, a 32-bit item must start on a byte boundary to satisfy this condition.
+        // This means we can just load the containing byte-aligned 32-bit word and extract the bits from there.
+        auto word_index = align_down_to(field.start_bit_index, 8) / 8;
```

Comment:
This is `auto word_index = ceil_div(field.start_bit_index, 8);` (also below)

Reply:
(For this line just dividing by 8 is actually enough. But the next line can be a `ceil_div`.)